lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -3063,18 +3069,25 @@ private function unregisterDocument($document)
 
     /**
      * @param object $document
-     * @param string $id       The document id to look for.
+     * @param string $pathOrUuid       The document id to look for.
      *
      * @return string generated object hash
      */
-    public function registerDocument($document, $id)
+    public function registerDocument($document, $pathOrUuid)
     {
         $oid = spl_object_hash($document);
-        $this->documentIds[$oid] = $id;
-        $this->identityMap[$id] = $document;
+        $this->documentIds[$oid] = $pathOrUuid;
+        $this->identityMap[$pathOrUuid] = $document;
 
         // frozen nodes need another state so they are managed but not included for updates
-        $frozen = $this->session->nodeExists($id) && $this->session->getNode($id)->isNodeType('nt:frozenNode');
+        try {
+            $node = $this->getNodeByPathOrUuid($pathOrUuid);
+            $frozen = $node instanceof  NodeInterface ? $node->isNodeType('nt:frozenNode') : false;
+        } catch (ItemNotFoundException $e) {
+            $frozen = false;
+        } catch (PathNotFoundException $e) {
+            $frozen = false;
```

Comment:
`$frozen = false` can be moved before the `try` block (doesn't need to be repeated twice)